vllm/utils.py

```diff
@@ -956,6 +956,39 @@ def find_nccl_library() -> str:
     return so_file
 
 
+prev_set_stream = torch.cuda.set_stream
+
+_current_stream = None
+
+
+def _patched_set_stream(stream: torch.cuda.Stream) -> None:
+    global _current_stream
+    _current_stream = stream
+    prev_set_stream(stream)
+
+
+torch.cuda.set_stream = _patched_set_stream
```

Comment:
It looks like we're not using `set_stream` anywhere in the vllm codebase. Could you add a unit test for this to make sure it's exercised?

Reply:
>  here we patch `torch.cuda.set_stream` to keep track of the current stream
    directly, so that we can avoid calling `torch.cuda.current_stream()`.

I might be confused about how `utils.current_stream()` works though

Reply:
`torch.cuda.graph` will call it internally to switch streams. so any test cases with cudagraph + nccl will test the PR's code.